### ISSUE TITLE
dmesg errors - upupbb - usb flash drive

### ISSUE THREAD

**Issue by peabee** [puppylinux-woof-CE/woof-CE#1252]
upupbb
BUILD_FROM_WOOF='master;ec80b41b;2018-06-23 11:36:40 +0800'
errors in dmesg:
udevd[9705]: Process '/sbin/pup_event_backend_modprobe' failed with exit code 1.
udevd[9872]: Process '/usr/local/pup_event/frontend_change add sdc1' failed with exit code 1.

also reports on the forum of problems:
http://murga-linux.com/puppy/viewtopic.php?p=1005534#1005534

**Comment by wdlkmpx** [comment]
I think pup_event_backend_modprobe should be removed, i think rerwin is working on this. But it's easy to delete it.. 30 seconds.

frontend_change: looks like a timing issue. In the testing branch, frontend_change doesn't 'fail with exit code 1'. But i think it fails anyway. I don't have that issue but i think i know why it happens.

- I think this issue is solved by adding a "sleep 1" or even "sleep 2" after "add)".

My first frontend_change implementation didn't have this issue, why? Because it relied completely on udev. But then the rox stuff and desktop options became a bit difficult to handle so the code was changing in one direction then again in another direction (to pave the way for the return of the compiled binary). I actually have a rewritten pup_event_frontend_d that doesn't depend on udev, unlike the old pup_event_frontend_d.c (it wasn't linked to udev but expected strings only udev produces).. but i feel that adding it would be a step backward.

In other words a huge amount of time was wasted just to keep the puppy tradition.

The current pup_event_frontend stuff has some other changes.

The project has tons of changes, even though it might seem that only a few changes happened. Just by reading the git log one can create a huge change log that can be posted to blog/forum.. it's as simple as that. Why am i refusing to do this? For several reasons.

I could merge testing to master today, then you build a iso and spend a week using it and asking a few people to test it privately before releasing it. ..  a new build with the current testing stuff should be advertised as a new version: version 18? Now version 19. Discourage upgrading unless you want some breakage to happen.

**Comment by peabee** [comment]
upupcc is the experimental testing branch build....it has the same dmesg reports....nobody should be "upgrading" from bb to cc.....

**Comment by wdlkmpx** [comment]
The frontend_backend code always produces errors

I don't  see frontend_startup error msgs in dpup. Even when
HOTPLUGNOISY is enabled (automatic launch of pmount).

The code changed a bit to supress the error msgs produced by udev when
frontend_change 'exists with error'.

In the testing branch there are (many) more changes in
pup_event_frontend and pmount

I added some more changes locally that make frontend_startup exit with
error only if probepart or probedisk fail.. exit 1  becomes exit 0.

But those error msgs don't cause the problem the user reported.

It's something deeper that might be already fixed, why don't you offer
him a private build of upupbb from the testing branch

The user hasn't posted the output of probedisk and probepart, might be
a discrepancy.. that is already fixed in the testing branch

Wish i had problematic hardware

**Comment by peabee** [comment]
upupcc
BUILD_FROM_WOOF='testing;04e0b628;2018-09-23 10:11:07 -0500'

error exit is line #54 of frontend_change:
udevd[9464]: Process '/usr/local/pup_event/frontend_change add sdb1' failed with exit code 14.

	case $ACTION in add|change)
		[ ! -e /sys/block/${DEV_NAME} ] && exit 14 ;;
	esac

**Comment by wdlkmpx** [comment]
It's 'exit 1'... are you using a modified frontend_change?

https://github.com/puppylinux-woof-CE/woof-CE/blob/testing/woof-code/rootfs-skeleton/usr/local/pup_event/frontend_change#L54

The error msgs i see happen with sdb1, sdc1... partitions... the script stops here:

https://github.com/puppylinux-woof-CE/woof-CE/blob/testing/woof-code/rootfs-skeleton/usr/local/pup_event/frontend_change#L25

`if ! fx_drv_is_ok $DEV_NAME ; then
	exit
fi`

**Comment by wdlkmpx** [comment]
[quote="musher0"]
What are we talking about in this thread? Authorship or ease of 
distribution? Those are two very different animals.
[/quote]

I wonder why this man always misses the point. 

It's about re-usability and distribution of code, about linking to libraries with incompatible licenses.

Nobody cares about copyrighted one-liners that somebody else already wrote.

[code]
The GPL is a communist license
[/code]

That's why it's called Copyleft. When you write GPLed software, you're donating it to mankind and denying businesses (and people who don't agree with donating your code to mankind or the FSF) the right to use it (unless they always provide the full source code), the right to modify it, and even link to your software, it must always be open source, in the terms defined by the Free Software Foundation and approved by R.S...

It's something for the common benefit (theoretically)

[code]
Not everything is black and white
[/code]

I'm not being racist..

People also use GPLed software to make money. That is, Red Hat. Their business model is to create complex GPL software, and they provide technical assistance and solutions for businesses.

Yes, Systemd is blessed.

[code]
The GPL was necessary
[/code]

The GPL made the Linux Kernel what it is now.

It might be a hateful license now, but at the time (i read) the wars about proprietary software were rampant.

That's why the BSDs lagged behind and now they have an offspring adopted by Apple.

[quote="technosaurus"][quote="torm"]Only for those who never read any license: 
it is not possible to "Stop using the GPL" on already GPL-ed software..  :wink:[/quote]Of course you can, you just implement the parts you need.
Anyone is free to implement anything they want from scratch for any reason they choose.
Rob Landley, the former maintainer of busybox (GPL), started toybox as a non-GPL replacement for busybox, core-utils and a few other projects.
[/url][/quote]

I read Rob started the infamous lawsuits against hardware vendors that used busybox. Then he realized his mistake and started toybox, now he hates the GPL and busybox.. he could have turned a blind eye in the first place..

Now he is android fan, or so it seems..

Anyway busybox is light years ahead of toybox, however musl is indeed in an excellent shape, i use it to compile static binaries.

[code]
Hedgehog dilemma
[/code]

As a writer of code and contributor to several projects and files, i find myself in a dilemma.

I've written a bunch of scripts and stuff, i also deleted many of them, i contributed to many GPLed scripts and apps.

I've rewritten many apps, but i've never added a license to my code.

I think i'll use the MIT license. But then again i remember that writing code has made a bitter person, and poorer.

And then the GPL mindset comes to mind, if i can't make money from my work then nobody can AHAHA!!

I don't know what to do.

**Comment by wdlkmpx** [comment]
Hahaha sorry, i pasted here a message that was meant for the puppy forum. Now deleted

**Comment by peabee** [comment]
I amended each exit 1 to give a different exit code....
there is no /sys/block/sdb1
only
/sys/block/sdb
exists

**Comment by wdlkmpx** [comment]
The script doesn't really fail when a partition is ignored, the whole drive will be processed by probepart/probedisk

In that case i just can't support upupbb and the lxpups

**Comment by peabee** [comment]
I don't understand......

**Comment by peabee** [comment]
- Your builds have modified core scripts
??????????????
Not modified by me...........

**Comment by wdlkmpx** [comment]
How come your frontend_change is different? Yeah i telepathically
changed it, it was me

https://github.com/puppylinux-woof-CE/woof-CE/issues/1252#issuecomment-424092815

**Comment by peabee** [comment]
I just did a local change for testing purposes......just to see which exit 1 was triggering - my builds are pure woof-ce!

**Comment by wdlkmpx** [comment]
It's simple, by doing a change your triggering the udev error msg.
With or without msg, the script fails for some reason because the
error happens somewhere else.

I can't reproduce the bug, it might be already fixed in testing, and i
already said what can be done for the time being

I'm going to merge testing to master

**Comment by wdlkmpx** [comment]
Ok, i apologise for bringing out the jerk in me .. again. Sorry..

Reading a bit more, it looks like a probepart issue, but i have no way of knowing without some debug info, however the testing branch does have some changes that might improve the situation... in fact  many changes to several key scripts. Porting all those changes to master is a tedious work and i will have to merge them back to testing.. creating duplicate commit messages or something.

Sometime ago someone reported a case where the 'full' blkid failed to identify a udf filesystem, 01micko then proceeded to add guess_fstype as a fallback and that worked. But now probepart is using busybox blkid.. and the fallback will be the 'full' blkid... guess_fstype is no more...

I'll build a upupbb iso from testing, tomorrow, and will send a PM to oldaolgeezer..

**Comment by peabee** [comment]
No problem..........
upupcc is a testing branch build almost identical to upupbb just using updated ubuntu components.....so you could download that:
https://sourceforge.net/projects/zestypup/files/Upup-Cosmic/Alpha/
you need to apply the alpha4 delta if you want the latest version of testing branch
BUILD_FROM_WOOF='testing;04e0b628;2018-09-23 10:11:07 -0500'

**Comment by wdlkmpx** [comment]
This file:
- http://01micko.com/wdlkmpx/woof-CE/busybox-1.29.3-x86_musl_static.pet
Should be uploaded to:
- http://distro.ibiblio.org/puppylinux/pet_packages-common32/

This file:
- http://01micko.com/wdlkmpx/woof-CE/busybox-1.29.3-x86_64_musl_static.pet
Should be uploaded to:
- http://distro.ibiblio.org/puppylinux/pet_packages-common64/

This file:
- http://01micko.com/wdlkmpx/woof-CE/busybox-1.29.3-arm_musl_static.pet
Should be uploaded to:
- http://distro.ibiblio.org/puppylinux/arm/pet_packages-armv6/

This file:
- http://01micko.com/wdlkmpx/woof-CE/busybox-1.29.3-aarch64_musl_static.pet
Is for the lolz

**Comment by peabee** [comment]
1st 3 done

**Comment by peabee** [comment]
BUILD_FROM_WOOF='testing;b795911a;2018-09-25 22:40:57 +0000'
https://sourceforge.net/projects/zestypup/files/Upup-Cosmic/Alpha
upupcc-18.09alpha5.iso

**Comment by peabee** [comment]
Other busybox also have _DEV and _DOC pets.....and thats what petbuild makes....
http://distro.ibiblio.org/puppylinux/pet_packages-slacko14.2/

**Comment by wdlkmpx** [comment]
upupcc looks a bit more stable, 

/etc/rc.d/rc.network_eth: line 46: ifplugstatus: command not found
/etc/rc.d/rc.network_eth: line 48: ifplugstatus: command not found
/etc/rc.d/rc.network_eth: line 50: ifplugstatus: command not found

https://packages.ubuntu.com/cosmic/i386/ifplugd/filelist

- yes|ifplugd|ifplugd|exe.dev.doc,nls

Issues with dbus

chgrp: invalid group: 'messagebus'
Starting system message bus: dbusdbus-daemon[4852]: Failed to start message bus: Failed to open "/run/dbus/pid": No such file or directory

/etc/init.d/messagebus must not exist. getting rid of that file in all builds will be another headache.

daemon.warn dbus-daemon[4852]: Failed to start message bus: Failed to open "/run/dbus/pid": No such file or directory

I still don't how to make dbus work, i remember compiling dbus for dpup stretch, but apparently it wasn't good to replace the original package.

- light browser causes a massive spike in cpu/ram usage (intel driver)

- palemoon updater doesn't work since version 28,,0.. it should be removed from the sfs.

etc

**Comment by peabee** [comment]
This is in builds both bb and cc:
:ifplugd:|pet|Packages-puppy-tahr-official|ifplugd-0.28-static|ifplugd|0.28-static||BuildingBlock|136K||ifplugd-0.28-static.pet||detect network active, compiled statically||||
